system/puppet.py

```diff
@@ -147,6 +156,7 @@ def main():
             facter_basename=dict(default='ansible'),
             environment=dict(required=False, default=None),
             certname=dict(required=False, default=None),
+            tags=dict(required=False, default=None),
```

Comment:
I recommend to use `type='list'`here. This would allow the use list style syntax:

~~~
puppet:
  tags:
    - update
    - nginx
~~~

but this still works as well

~~~
puppet: tags=update,nginx
~~~

I usually also add an alias `aliases=['tag']`, so this would also work:

~~~
puppet:
  tag: update
~~~